engine/src/main/java/org/terasology/identity/ClientIdentity.java

```diff
@@ -49,4 +50,21 @@ public PrivateIdentityCertificate getPlayerPrivateCertificate() {
         }
         return playerPrivateCertificate;
     }
+
+    @Override
+    public int hashCode() {
+        return playerPrivateCertificate.hashCode() ^ playerPrivateCertificate.hashCode();
```

Comment:
Also changed the hashCode() implementation in `PrivateIdentityCertificate` to use this method. I'm not sure if I should change it in `PlayerPublicCertificate` too since an implementation was already there (I didn't added it in this PR) which simply returns `signature.hashCode()`; combining with the other fields (id, modulus and exponent) will probably reduce the collision rate but could also slow down the computation of the hash, what do you think about it?